resources/lib/sony.py

```diff
@@ -1,415 +1,414 @@
-        dialog = xbmcgui.Dialog()
-        remove = dialog.yesno(location,msg)
-        if remove:
-            r = requests.delete(url, headers=headers, cookies=self.load_cookies(), data=payload, verify=self.verify)
-
-            if r.status_code == 200:
-                self.notification_msg("Success!", "Removed from "+location)
-            else:
-                self.notification_msg("Fail", "Not added")
-
-
-    def put_resume_time(self, airing_id, channel_id, program_id, series_id, tms_id):
-        url = self.user_action_url+'/watch_history'
-        headers = {"Accept": "*/*",
-                   "Content-type": "application/json",
-                   "Origin": "https://themis.dl.playstation.net",
-                   "Accept-Language": "en-US",
-                   "Referer": "https://themis.dl.playstation.net/themis/destro/4-4-5a/",
-                   "Accept-Encoding": "gzip, deflate",
-                   "User-Agent": self.ua_android_tv,
-                   "Connection": "Keep-Alive",
-                   'reqPayload': self.addon.getSetting(id='reqPayload'),
-                   'X-Requested-With': 'com.snei.vue.android'
-                   }
-
-        payload = '{"series_id":'+series_id+','
-        payload += '"program_id":'+program_id+','
-        payload += '"channel_id":'+channel_id+','
-        payload += '"tms_id":"'+tms_id+'",'
-        payload += '"airing_id":'+airing_id+','
-        payload += '"last_watch_date":"2017-08-11T00:40:43Z",'
-        payload += '"last_timecode":"00:01:46",'
-        payload += '"start_timecode":"00:00:00:00",'
-        payload += '"fully_watched":false,'
-        payload += '"stream_type":"dvr"}'
-
-        #xbmc.log(payload)
-        r = requests.put(url, headers=headers, data=payload, verify=self.verify)
-
-
-    def save_cookies(self, cookiejar):
-        addon_profile_path = xbmc.translatePath(self.addon.getAddonInfo('profile'))
-        cookie_file = os.path.join(addon_profile_path, 'cookies.lwp')
-        cj = cookielib.LWPCookieJar()
-        try:
-            cj.load(cookie_file,ignore_discard=True)
-        except:
-            pass
-        for c in cookiejar:
-            args = dict(vars(c).items())
-            args['rest'] = args['_rest']
-            del args['_rest']
-            c = cookielib.Cookie(**args)
-            cj.set_cookie(c)
-        cj.save(cookie_file, ignore_discard=True)
-
-
-    def load_cookies(self):
-        addon_profile_path = xbmc.translatePath(self.addon.getAddonInfo('profile'))
-        cookie_file = os.path.join(addon_profile_path, 'cookies.lwp')
-        cj = cookielib.LWPCookieJar()
-        try:
-            cj.load(cookie_file, ignore_discard=True)
-        except:
-            pass
-
-        return cj
-
-
-    def notification_msg(self, title, msg):
-        dialog = xbmcgui.Dialog()
-        dialog.notification(title, msg, xbmcgui.NOTIFICATION_INFO, 5000)
+        self.addon.setSetting(id='EPGreqPayload', value='')

```

Comment:
I'm not sure I understand why you've created a new variable name for the reqpayload?

Reply:
There are two variables for reqpayload. I believe you had reqpayload reauth if it was already in the addon settings and then update reqpayload to the new one. Either the reauth wasn't working or sony changed how reqoayload was used. I only noticed in the web console when logging in and viewing EPG info.
Basically the initial reqpayload did not allow the epg url to load the json info. Getting the second reqpayload from the set_profile url resolved that. Saving both allows authorization on one and EPG info on The other. Hope that helps explain everything.

Reply:
Yes, that makes sense. Reauth was kind of my interpretation of what "should" happen so it could be an incorrect assumption.

Reply:
I just realized the Live Tv section won't load in my last commit because the json info for channels is different than shows. Live TV json info is listed the same as shows. So i changed live tv url to list_shows instead of list_channels. I haven't made a commit yet because you're going crazy looking at all my other commits.

Reply:
It's a good problem to have 😄. I'd pull the previous changes in but there seems to conflict with the sony.py file.

Reply:
I thought I fixed the conflict yesterday, but it's fixed now. From now on I will edit my tested code on github instead of uploading the file again. It didn't really say why there was a conflict except the code was double.